go/service/crypto.go

```diff
@@ -7,30 +7,84 @@ import (
 	"github.com/keybase/client/go/engine"
 	"github.com/keybase/client/go/libkb"
 	keybase1 "github.com/keybase/client/go/protocol"
-	rpc "github.com/keybase/go-framed-msgpack-rpc"
 	"golang.org/x/net/context"
 )
 
 type CryptoHandler struct {
-	*BaseHandler
 	libkb.Contextified
 }
 
-func NewCryptoHandler(xp rpc.Transporter, g *libkb.GlobalContext) *CryptoHandler {
+func NewCryptoHandler(g *libkb.GlobalContext) *CryptoHandler {
 	return &CryptoHandler{
-		BaseHandler:  NewBaseHandler(xp),
 		Contextified: libkb.NewContextified(g),
 	}
 }
 
+func (c *CryptoHandler) getDelegatedSecretUI() libkb.SecretUI {
+	if c.G().UIRouter == nil {
+		c.G().Log.Debug("no UIRouter to get SecretUI from")
+		return nil
+	}
+
+	ui, err := c.G().UIRouter.GetSecretUI()
+	if err != nil {
+		c.G().Log.Debug("UIRouter.GetSecretUI() returned %v", err)
+		return nil
+	}
+
+	if ui == nil {
```

Comment:
This can happen, too. From looking at the code, this is expected, so added a comment in the UIRouter interface, too.